src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -259,26 +261,27 @@ trait Erasure {
     if (sym != NoSymbol && sym.enclClass.isJavaDefined)
       erasure(sym)(tp)
     else if (sym.isClassConstructor)
-      specialConstructorErasure(sym.owner, tp)
-    else
-      specialScalaErasure(tp)
+      specialConstructorErasure(sym.owner, sym, tp)
+    else {
+      specialScalaErasureFor(sym)(tp)
+    }
 
-  def specialConstructorErasure(clazz: Symbol, tpe: Type): Type = {
+  def specialConstructorErasure(clazz: Symbol, ctor: Symbol, tpe: Type): Type = {
```

Comment:
it seems `ctor` is only used for `ctor.isScala3Defined`, and that should always be the same as `clazz.isScala3Defined`, no? then we could do without that new parameter.

Reply:
Good catch thank you, this was an artifact of when I was only using attachments on methods to encode Scala 3-ness